doc/sphinx/practical-tools/coq-commands.rst

```diff
@@ -42,16 +42,52 @@ toplevel with the command ``Coqloop.loop();;``.
 Batch compilation (coqc)
 ------------------------
 
-The ``coqc`` command takes a name *file* as argument. Then it looks for a
-file named *file*.v, and tries to compile it into a
-*file*.vo file (See :ref:`compiled-files`).
-
-.. caution::
-
-   The name *file* should be a regular Coq identifier as defined in Section :ref:`lexical-conventions`.
-   It should contain only letters, digits or underscores (_). For example ``/bar/foo/toto.v`` is valid,
-   but ``/bar/foo/to-to.v`` is not.
-
+The ``coqc`` command compiles a Coq proof script file with a ".v" suffix
+to create a compiled file with a ".vo" suffix.  (See :ref:`compiled-files`.)
+The components of the filename must be valid Coq identifiers as described in
+:ref:`lexical-conventions`; they should contain only letters, digits or
+underscores (_) with a ".v" suffix on the final component.
+For example ``/bar/foo/toto.v`` is valid, but ``/bar/foo/to-to.v`` is not.
+
+We recommend specifying a logical directory name (which is also the module name)
+with the `-R` or the `-Q` options.
+Generally we recommend using provided utilities such as `make` to build Coq projects.
+See :ref:`building_coq_project`.
+
+.. example:: Compiling and loading a single file
+
+   If `foo.v` is in Coq's current directory, you can use `coqc foo.v`
+   to compile it and then `Require foo.` in your script.  But this
+   doesn't scale well for larger projects.
+
+   Generally it's' better to define a new module:
+   To compile `foo.v` as part of a module `Mod1` that is rooted
+   at `.` (i.e. the directory containing `foo.v`), run `coqc -Q . Mod1 foo.v`.
+
+   To make the module available in `CoqIDE`, include the following lines in the
+   `_CoqProject` file (see :ref:`coq_makefile`) in the directory from which you
+   start `CoqIDE`.  For now, you need to close and reload a named
+   script file for `CoqIDE` to pick up the change or restart `CoqIDE`.
+   The project file name is configurable in `Edit / Preferences / Project`.
+
+      .. coqdoc::
+         -R ../empty_dir/mod1 Mod1
+         ./foo.v
+
+   It's also possible to load a module within `coqtop` or `coqide` with
+   these commands like this.  The drawback of this is that it adds
+   environment-specific information (the PATH) to your script, making
+   it non-portable.
```

Comment:
```suggestion
   it non-portable. Therefore, this is discouraged.
```

Reply:
It could also be the case that these commands get removed in the future, as they commit to a library organization model which may not hold. Personally, I wouldn't recommend them under any circumstance.

Reply:
We document Coq as it is, not as we want it to become.  Only by mentioning and explaining this feature can we say not to use it.  Feel free to start a discussion on removing it--though I think we should review how to use the various compilation options first and perhaps tweak them.